ocs_ci/ocs/ocp.py

```diff
@@ -874,3 +875,104 @@ def verify_images_upgraded(old_images, object_data):
         f"All the images: {current_images} were successfully upgraded in: "
         f"{name}!"
     )
+
+
+def check_upgrade_completed(target_version):
+    """
+    Check if OCP upgrade process is completed:
+        checks if all ClusterOperator images updated
+
+    Args:
+        target_version (str): expected OCP client
+
+    Returns:
+        bool: True if success, False if failed
+
+    """
+    log.info(f"target_version=: {target_version}")
+    log.info(f"current OCP version is: {get_current_oc_version()}")
+    cur_version = get_current_oc_version()
+    if cur_version == target_version or target_version.startswith(cur_version):
+        log.info(f"cluster operator upgrade to build {target_version} completed")
+        return True
+
+    log.debug(f"upgrade not completed")
+    return False
+
+
+def upgrade_ocp(image_path, image):
+    """
+    upgrade OCP version
+
+    Args:
+        image (str): image to be installed
+        image_path (str): path to image
+
+    """
+    ocp_o = OCP()
+    ocp_o.exec_oc_cmd(
+        f"adm upgrade --to-image={image_path}:{image} "
+        f"--allow-explicit-upgrade --force "
+    )
+    log.info(f"Upgrading OCP to version: {image}")
+
+
+def get_current_oc_version():
```

Comment:
don't we have a similar function already?
I'm almost sure we use similar functionality in the deployment phase but might be wrong
@petr-balogh could you keep me honest here?

Reply:
I do not remember such function.

We have just some helper function which deal with version of OCP from config under utils called: get_ocp_version

But this one is different.